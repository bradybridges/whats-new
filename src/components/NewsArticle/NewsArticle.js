import React from 'react';
import './NewsArticle.scss';

class NewsArticle extends React.Component {

  render() {
    return (
      <article className="newsArticle">
        <img src={this.props.news.img} alt="news" />
        <h3>{this.props.news.headline}</h3>
        <p>{this.props.news.description}</p>
        <form action={this.props.news.url} method="get" target="_blank">
         <button className="article-button" type="submit">Read Full Article</button>
        </form>
      </article>
    )
  }
}

export default NewsArticle; 