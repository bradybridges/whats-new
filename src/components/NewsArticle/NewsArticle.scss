$primary-color: #3f4446;
$secondary-color: #e3eef0;

img {
  border-radius: 5px;
  width: 100%;
}

h3 {
  font-size: 2em;
  margin: 20px 0px;
  text-align: center;
  width: 80%;
}

p {
  font-size: 1.3em;
  margin-bottom: 20px;
  text-align: center;
  width: 75%;
}

.newsArticle {
  align-items: center;
  background-color: #1f1d21f2;
  border: solid 1px #e3eef06e;
  border-radius: 10px;
  box-shadow: 8px -6px 20px 0px #dae7ed40;
  color: $secondary-color;
  display: flex;
  flex-direction: column;
  margin: 20px auto;
  padding: 25px;
  width: 30%;
}

.article-button {
  background-color: $primary-color;
  border-radius: 4px;
  color: $secondary-color;
  font-family: 'Roboto Slab', serif;
  font-size: 1.2em;
  height: 50px;
  margin-left: 30px;
  transition: .3s;
  width: 300px;
  &:hover {
    background-color: #1b1a1c;
    border: solid 2px #779c43;
    color: #779c43;
    cursor: pointer;
    transform: scale(1.1);
  }
}

@media only screen and (max-width: 1000px) {
  .newsArticle {
    width: 45%;
  }
}

@media only screen and (max-width: 850px) {
  .newsArticle {
    padding: 5px;
    width: 90%;
  }

  .article-button {
    margin-left: 0;
  }
}

@media only screen and (max-width: 500px) {
  h3, p {
    width: 100%;
  }
  .article-button {
    width: 275px;
  }
}