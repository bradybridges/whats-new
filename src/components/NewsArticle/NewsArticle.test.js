import React from 'react';
import NewsArticle from './NewsArticle';
import { shallow } from 'enzyme';

describe('NewsArticle', () => {
  
  it('should match the snapshot', () => {
    const newsArticleMock = {headline: 'Hello', description:'This is the description.', id: 1};
    const wrapper = shallow(<NewsArticle key={1} news={newsArticleMock} />);
    expect(wrapper).toMatchSnapshot();
  })
})