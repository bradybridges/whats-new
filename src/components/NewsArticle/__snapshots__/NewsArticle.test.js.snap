// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot 1`] = `
<article
  className="newsArticle"
>
  <img
    alt="news"
  />
  <h3>
    Hello
  </h3>
  <p>
    This is the description.
  </p>
  <form
    method="get"
    target="_blank"
  >
    <button
      className="article-button"
      type="submit"
    >
      Read Full Article
    </button>
  </form>
</article>
`;
